### DISCUSSION TITLE
Klonen der Repository

### DISCUSSION THREAD

**Discussion by CBRocker** [tbnobody/OpenDTU#88] [Q&A]
Hallo,
ich habe seit kurzem ein BKW mit Hoymiles 600.
Nun möchte ich natürlich die Daten auslesen. Ich hatte dafür die Wahl zwischen Ahoy oder OpenDTU. Habe mich für OpenDTU entschieden, aber ich hänge leider schon ganz am Anfang. Ich bin zwar in der IT-Branche tätig, aber mit Microcontrollern, deren Programmierung, Flashen usw. hatte ich noch nie was zu tun.
Ich bitte um Hilfe beim Einrichten von OpenDTU.

Ich habe diese Hardware: ESP32-WROOM-32 und nRF24L01+PA+LNA

Ich habe Visual Studio Code + Extension PlatformIO installiert.

Und nun hänge ich.
Muß ich in PlatformIO ein Projekt anlegen? Wenn ja, mit welchen Parametern?
Wie wird das Repository geklont?
Die Hardware ist auf COM Port 3, welche Einträge sind in der platformio.ini dafür zu machen?

Ich denke, wenn ich diese Fragen beantwortet bekomme, komme ich wohl weiter.

Ich freue mich auf Hilfe. Wie gesagt, was dieses Thema angeht, bin ich absoluter Anfänger.

Vielen Dank,
Grüße Christoph

**Comment by sivar2311** [comment]
Hallo @CBRocker 

- In VS-Code die Command-Palette mit `STRG` + `SHIFT` + `P` öffnen
- `git clone` eingeben und anklicken
- Die Repository-URL `https://github.com/tbnobody/OpenDTU` eingeben bzw. Einfügen
- Zielverzeichnis  auswählen (ggf. Erstellen)
- Das geklonte Repository öffnen
- ggf. platformio.ini anpassen (bei mir z.B. COM10)
- Upload...fertig 

Eventuell lädt PlatformIO noch benötigte Dateien herunter

Und hier das ganze einmal in einem [Video](https://youtu.be/9cA_esv3zeA)

**Reply by CBRocker** [reply]
Hallo @sivar2311 

vielen Dank für die Infos. Ich habe mich ziemlich eingelesen und das mittlerweile auch herausgefunden mit der Command-Palette.
Aber wenn ich da git clone eingebe, passiert gar nichts "keine übereinstimmenden Befehle".
Ich habe VS Code, dazu die deutsche Sprache dafür installiert und PlattformIO IDE.

Mir scheint, bei mir fehlt noch was?
![git_clone](https://user-images.githubusercontent.com/96444759/185476107-d78384b1-d500-4e01-83d2-6ce39cd56835.jpg)

**Reply by sivar2311** [reply]
Hast du GIT auf deinem PC installiert?

**Reply by CBRocker** [reply]
Nein. Das ist auch Voraussetzung? GIT für Windows?

**Reply by sivar2311** [reply]
Ich bin mir nicht ganz 100%ig sicher aber ich denke schon (ist schon so lange her das ich alles bei mir eingerichtet habe).
Die Windows-Version gibt es unter https://git-scm.com/download/win

**Reply by CBRocker** [reply]
habe es nun installiert. Jetzt sieht es anders aus. Das ist absolutes Neuland für mich. Ich spiele zwar schon länger mit Rasperry PI, iobroker wg. SmartHome rum und hatte deswegen mit Git Hub schon Berührungen, aber Microcontroller ist neu für mich :-)
Ich danke dir für die große Hilfe. Jetzt kann ich weiter experimentieren.

**Reply by sivar2311** [reply]
Und hier gibt's noch eine Anleitung https://www.jcchouinard.com/install-git-in-vscode/

**Reply by sivar2311** [reply]
Super, das es geklappt hat :)
Dann ergänzen wir uns ganz gut. 
Ich kenne mich wiederum mit RasperryPI und iobroker nicht aus ;)
Dafür aber ein bisschen mit VS-Code, Arduino (ESP8266 und ESP32)

**Reply by CBRocker** [reply]
SUCCESS :-)

Super, daß du mir so klasse und schnell geholfen hast.
Und danke für die Anleitung und das YT-Video. Die Anleitung lese ich mir durch.
Jetzt habe ich eine weitere Sucht :-)

SmartHome mit iobroker, Shellys, HmIP usw. macht schon süchtig.
Und jetzt noch Microcontroller. Bin ich mal gespannt, ob ich den Hoymiles auslesen kann.
Mit den ESP8266 und ESP32 gibt es ja unbegrenzte Möglichkeiten. Immerhin habe ich dank dir
nun ca. 0,00001 % Wissen bei VSCode und ESP´s.

Besten Dank dir, 
schönes Wochenende

Grüße aus Oberndorf am Neckar

**Comment by madmartin** [comment]
Vielleicht kann @tbnobody  mal das Wiki hier im Repository freischalten, da gehören solche Anleitungen hinein!

**Reply by sivar2311** [reply]
Hallo @madmartin! Vielleicht kannst du den Punkt auch noch in der [README.md](https://github.com/tbnobody/OpenDTU/blob/master/README.md#flashing-and-starting-up), jeweils vor den Punkten `Clone this repository` den Punkt `Install git` hinzufügen. Der Vollständigkeit halber gehört es auch dort hinein und wird wahrscheinlich eher gefunden als im Wiki.
